### PR TITLE
Add group_by_use_nulls setting

### LINKED ISSUE #37359: A setting `rollup_use_nulls` for standard compatibility
**Use case**

When it is set and GROUP BY has WITH ROLLUP, WITH CUBE, or GROUPING SETS, the columns from GROUP BY are interpreted as Nullable.

### PR DESCRIPTION
**Pull request by novikd**
### Changelog category (leave one):

- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add `group_by_use_nulls` setting to make aggregation key columns nullable in the case of ROLLUP, CUBE and GROUPING SETS. Closes #37359

### CODE REVIEW

**Comment by tavplubix** [comment]
Consider adding it here:
https://github.com/ClickHouse/ClickHouse/blob/e307a728caaa36c9d603422152c71850a602bacf/docker/test/stress/stress#L11

File: src/Processors/Transforms/RollupTransform.cpp [hunk 1]
```diff
@@ -1,36 +1,83 @@
...
     , params(std::move(params_))
-    , aggregates_mask(getAggregatesMask(params->getHeader(), params->params.aggregates))
+    , use_nulls(use_nulls_)
 {
     keys.reserve(params->params.keys_size);
     for (const auto & key : params->params.keys)
         keys.emplace_back(input.getHeader().getPositionByName(key));
+
+    intermediate_header = getOutputPort().getHeader();
+    intermediate_header.erase(0);
```
Comment (CurtizJ) [inline src/Processors/Transforms/RollupTransform.cpp:21]: Please, explain why first column is removed.

Comment (novikd) [reply src/Processors/Transforms/RollupTransform.cpp:21]: First column is `__grouping_set`. It's added exactly before returning generated chunk. `intermediate_heade` is used to correctly create columns with defaults for input blocks (they don't have `__grouping_set` column).

**Comment by CurtizJ** [comment]
Stress test failures are related:
```sql
SELECT
    CounterID AS k,
    quantileBFloat16(0.5)(ResolutionWidth)
FROM remote('127.0.0.{1,2}', default, hits)
GROUP BY k
ORDER BY
    count() DESC,
    CounterID ASC
LIMIT 10
SETTINGS group_by_use_nulls = 1

Query id: 3efcd4cf-74fc-4e78-b39b-94d1227953c7

→ Progress: 8.87 million rows, 53.24 MB (83.89 million rows/s., 503.35 MB/s.)  99%
0 rows in set. Elapsed: 0.190 sec. Processed 8.87 million rows, 53.24 MB (46.81 million rows/s., 280.85 MB/s.)

Received exception from server (version 22.7.1):
Code: 49. DB::Exception: Received from localhost:9000. DB::Exception: Bad cast from type DB::ColumnNullable to DB::ColumnVector<unsigned int>. (LOGICAL_ERROR)
```

**Review by CurtizJ — APPROVED** [review]

**Pull request merged**